third-party/realdds/include/realdds/dds-control-server.h

```diff
@@ -0,0 +1,47 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2022 Intel Corporation. All Rights Reserved.
+
+#pragma once
+
+
+#include <realdds/topics/control/controlPubSubTypes.h>  // raw::device::control
+#include <librealsense2/utilities/concurrency/concurrency.h>
+
+#include <memory>
+#include <string>
+
+
+namespace realdds {
+
+
+namespace topics {
+namespace raw {
+namespace device {
+class control;
+}
+}  // namespace raw
+}  // namespace topics
+
+
+class dds_subscriber;
+class dds_topic_reader;
+
+
+// A control is a command sent from a client (camera user) and the server (this class) receives and interprets.
+//
+class dds_control_server
```

Comment:
From my understanding of the usage, this is not really a server but a client -- a listener for control messages.

If so -- then how is this different than a topic-reader? Do you envision more logic being placed here? I don't see much logic belonging in realdds other than high-level commands (stream start/stop, control get/set) that would still need customization by the end implementation.

Reply:
I think that we will need more logic to handle stuff like query-reply pairs and msg-id matching.
If it will all be in the application than you are right and we can refactor this out

Reply:
I think it'll be where the control messages are handled.
Most controls will be handled in the app and only a few will be done entirely in realdds and even then it'll likely be in the device-server. If there's a lot of code we can certainly push to another class but at this time I don't see much justification.